### PR TITLE
DAC6-1296: Match to correct error status and log the actual error status

### PR DESCRIPTION
**Pull request by BriWak**

### CODE REVIEW

File: app/controllers/ContactPhoneController.scala [hunk 1]
```diff
@@ -55,36 +53,26 @@ class ContactPhoneController @Inject() (
...
-          "form"      -> request.userAnswers.get(ContactPhonePage).fold(form)(form.fill),
-          "regime"    -> regime.toUpperCase,
-          "name"      -> name,
-          "pageTitle" -> s"contactPhone.title.$suffix",
-          "heading"   -> s"contactPhone.heading.$suffix",
-          "hintText"  -> hintWithNoBreakSpaces(),
-          "action"    -> routes.ContactPhoneController.onSubmit(mode, regime).url
-        )
-    }
+    val name         = request.userAnswers.get(ContactNamePage).orElse(orIndividual)
```
Comment (ashwinirudrappa) [inline app/controllers/ContactPhoneController.scala:60]: @BriWak It's an existing code, We should move away from orElse and place the correct condition
Can we fix this

File: app/controllers/IsContactTelephoneController.scala [hunk 2]
```diff
@@ -55,36 +53,26 @@ class IsContactTelephoneController @Inject() (
...
-          "regime"    -> regime.toUpperCase,
-          "name"      -> name,
-          "pageTitle" -> s"isContactTelephone.title.$suffix",
-          "heading"   -> s"isContactTelephone.heading.$suffix",
-          "action"    -> routes.IsContactTelephoneController.onSubmit(mode, regime).url,
-          "radios"    -> Radios.yesNo(filledForm("value"))
-        )
-    }
+    val name         = request.userAnswers.get(ContactNamePage).orElse(orIndividual)
+    val filledForm   = request.userAnswers.get(IsContactTelephonePage).fold(form)(form.fill)
```
Comment (ashwinirudrappa) [inline app/controllers/IsContactTelephoneController.scala:59]: Same as above, orElse

**Review by ashwinirudrappa — COMMENTED** [review]
LGTM, only one thing needs updating it is ApiError
https://github.com/hmrc/register-for-exchange-of-information-frontend/blob/86238ec0b1dca6577c7c3bf0331ecb6093e7421e/app/models/error/ApiError.scala#L39

We need to separate out SERVICE_UNAVAILABLE        = 503

**Review by soljeggers — APPROVED** [review]
LGTM

**Review by soljeggers — APPROVED** [review]

**Review by ashwinirudrappa — APPROVED** [review]

**Pull request merged**